File: app/src/main/java/com/jarvis/app/ai/ResultComposer.java
package com.jarvis.app.ai;

public class ResultComposer {

    public String composeResult(String geminiResult, String chatgptResult, String grokResult) {
        StringBuilder finalResult = new StringBuilder();

        if (geminiResult != null && !geminiResult.isEmpty()) {
            finalResult.append("Planning: ").append(geminiResult).append("\n\n");
        }
        if (chatgptResult != null && !chatgptResult.isEmpty()) {
            finalResult.append("Personalization: ").append(chatgptResult).append("\n\n");
        }
        if (grokResult != null && !grokResult.isEmpty()) {
            finalResult.append("Analysis: ").append(grokResult).append("\n");
        }

        if(finalResult.length() == 0){
            return "I was unable to process that request.";
        }

        return finalResult.toString().trim();
    }

     public String parseGeminResponse(String responseBody){
         // Basic parsing, would need to be more robust for production
         return responseBody;
     }

     public String parseChatGPTResponse(String responseBody){
         // Basic parsing
         return responseBody;
     }

     public String parseGrokResponse(String responseBody){
        // Basic parsing
        return responseBody;
     }
}

File: jarvis_flutter/lib/ai/result_composer.dart
class ResultComposer {
  String composeResult({
    String? geminiResult,
    String? chatgptResult,
    String? grokResult,
  }) {
    final buffer = StringBuffer();

    if (geminiResult != null && geminiResult.isNotEmpty) {
      buffer.writeln("Planning: $geminiResult\n");
    }
    if (chatgptResult != null && chatgptResult.isNotEmpty) {
      buffer.writeln("Personalization: $chatgptResult\n");
    }
    if (grokResult != null && grokResult.isNotEmpty) {
      buffer.writeln("Analysis: $grokResult");
    }

    if (buffer.isEmpty) {
      return "I was unable to process that request.";
    }

    return buffer.toString().trim();
  }
}
